### PR TITLE
fix: make getResponsiveSteps getter work properly

### PR DESCRIPTION
**Pull request by akhodyko**
## Description
Form layout responsive steps getter does not work at all.
Included test describes the case.

## Type of change

- [x] Bugfix
- [ ] Feature

### CODE REVIEW

**Review by vursen — CHANGES_REQUESTED** [review]
Thank you for your contribution. Please note that ResponsiveStep can be also initialized without minWidth, and this currently results in an exception. This case needs to be addressed as well.

P.S I'm a bit surprised that ResponsiveStep doesn't provide any getters, which means getResponsiveSteps has very limited practical value, as it only provides information on the number of added steps.

**Comment by yuriy-fix** [comment]
Dear @akhodyko, the API has been confirmed to not be working properly and it should be fixed. Could you please share your use case for the API usage, so enhancements can be considered along the way and corresponding issues can be created?

File: vaadin-form-layout-flow-parent/vaadin-form-layout-flow/src/main/java/com/vaadin/flow/component/formlayout/FormLayout.java [hunk 1]
```diff
@@ -115,6 +116,9 @@ public String toString() {
         private int columns;
         private LabelsPosition labelsPosition;
 
+        public ResponsiveStep() {
```
[the remark this answers is not in the document]
Comment (akhodyko) [reply vaadin-form-layout-flow-parent/vaadin-form-layout-flow/src/main/java/com/vaadin/flow/component/formlayout/FormLayout.java]: it's possible to make this constructor private and replace a 
```JsonSerializer.toObjects(ResponsiveStep.class, stepsJsonArray);```
with
```
List<ResponsiveStep> steps = new ArrayList<>();
for (int i = 0; i < stepsJsonArray.length(); i++) {
    steps.add(new ResponsiveStep().readJson(stepsJsonArray.get(i)));
}
return steps;
```

File: vaadin-form-layout-flow-parent/vaadin-form-layout-flow/src/test/java/com/vaadin/flow/component/formlayout/tests/FormLayoutTest.java [hunk 2]
```diff
@@ -82,4 +83,20 @@ public void verifyColspanCodeBehaviour() {
 
     }
 
+    @Test
+    public void setResponsiveSteps_getResponsiveSteps() {
+        FormLayout formLayout = new FormLayout();
+
+        formLayout.setResponsiveSteps(new ResponsiveStep(null, 1));
```
[the remark this answers is not in the document]
Comment (akhodyko) [reply vaadin-form-layout-flow-parent/vaadin-form-layout-flow/src/test/java/com/vaadin/flow/component/formlayout/tests/FormLayoutTest.java:90]: done

**Comment by vursen** [comment]
We had an internal discussion and decided to reuse the existing public constructor instead of introducing a new private one. We can always add the default constructor later if there is a request for it. I've updated the PR accordingly.

**Review by vursen — APPROVED** [review]

**Pull request merged**